Remove destruction of WiFi interface in test runs

SeppoTakalo opened the pull request:
## Description

Most of our IP stacks don't allow removal of interfaces so
interface destructor can not reliably clean up. Therefore we
cannot rely its behaviours in test case.

Instead run interface->disconnect() in case interface was already
created.


## Status

**READY**

## Migrations

NO

0xc0170 commented:
@SeppoTakalo Can you please rebase from latest master please, will fix travis timeout.

betzw commented:
Isn't this somehow changing the behavior of the tests in the sense that disconnecting - as far as I understand - should maintain previously set credentials?!?

SeppoTakalo commented:
Rebased.

@betzw Very valid point.
The testcase `WIFI-CONNECT-NOCREDENTIALS` actually depends on the fact that no credentials have been fed to the driver. Therefore I now moved it to be very first test case after a constructor.

0xc0170 commented:
/morph build

0xc0170 commented:
/morph build

cmonr commented:
/morph test

cmonr commented:
/morph test

cmonr commented:
/morph test
